### PR TITLE
Add acceptance and unit testing

### PR DESCRIPTION
**Pull request by mscottford**
I have some new features that I'd like to add, and I wanted to make sure I kept all existing behavior in the process, so I added test coverage as a first step. This pull request brings the line coverage to ~85% and the branch coverage to about ~70%. Coverage is measured via `simplecov` integration and can be cleanly generated by running `bundle exec rake`.

### CODE REVIEW

**Comment by mscottford** [comment]
In response to the DCO failure, I'll be signing all of my commits and doing a force push after that's done.

**Comment by stevespringett** [comment]
Big thanks for the PR. I had to update the GitHub actions that were used. The ones specified were deprecated and no longer supported. Could you sync your fork to master? I'd like to get a passing build before merge.

**Comment by mscottford** [comment]
> Big thanks for the PR. I had to update the GitHub actions that were used. The ones specified were deprecated and no longer supported. Could you sync your fork to master? I'd like to get a passing build before merge.

Not a problem! I've rebased off of `master` and I've signed my commits.

**Comment by mscottford** [comment]
I see that CI is still failing. I'm going to work on getting that to pass. I'm going to add a matrix of ruby versions to test against.

**Comment by mscottford** [comment]
> I see that CI is still failing. I'm going to work on getting that to pass. I'm going to add a matrix of ruby versions to test against.

CI appears to have been failing because `activesupport` version 7.x does not support Ruby versions older than 2.7, so I removed Ruby 2.6 from the support matrix. If Ruby 2.6 needs to be supported, then we'll need to depend on an older version of `activesupport`.

File: cyclonedx-ruby.gemspec [hunk 1]
```diff
@@ -8,14 +8,28 @@ Gem::Specification.new do |spec|
   spec.description = 'CycloneDX is a lightweight software bill-of-material (SBOM) specification designed for use in application security contexts and supply chain component analysis. This Gem generates CycloneDX BOMs from Ruby projects.'
   spec.authors     = ['Joseph Kobti', 'Steve Springett']
   spec.email       = 'josephkobti@outlook.com'
-  spec.files       = ['lib/bom_builder.rb', 'lib/bom_helpers.rb', 'lib/licenses.json', 'lib/bom_component.rb']
   spec.homepage    = 'https://github.com/CycloneDX/cyclonedx-ruby-gem'
   spec.license     = 'Apache-2.0'
-  spec.executables << 'cyclonedx-ruby'
+
+  spec.required_ruby_version = ">= 2.7.0"
```
[the remark this answers is not in the document]
Comment (mscottford) [reply cyclonedx-ruby.gemspec:14]: This has been fixed in f510a03.

**Comment by mscottford** [comment]
Yay! That's a passing build, @stevespringett. Please let me know if you have any code review feedback.

**Comment by stevespringett** [comment]
Fantastic. Thanks so much for the PR.

**Pull request merged**